base/arraymath.jl

```diff
@@ -67,15 +60,21 @@ end
 
 for f in (:.+, :.-, :.*, :./, :.\, :.^, :.÷, :.%, :.<<, :.>>, :div, :mod, :rem, :&, :|, :$)
     @eval begin
-        function ($f){T}(A::Number, B::AbstractArray{T})
-            F = similar(B, promote_array_type($f,typeof(A),typeof(B)))
+        function ($f)(A::Number, B::AbstractArray)
+            P = promote_op($f, typeof(A), eltype(B))
+            T = promote_array_type($f, typeof(A), eltype(B), P)
+            T === Any && return [($f)(A, b) for b in B]
+            F = similar(B, T)
```

Comment:
If `T==Any`, it will return an `Array`, otherwise it may not, depending on `similar`. But maybe we don't have to worry about that now. These changed will be obsoleted by the upcoming lowering of `a .op b` as `(op).(a,b)` anyway.